content/6_Advanced/String_Search.mdx

```diff
@@ -139,6 +138,46 @@ but instead finds the longest common prefix of a string and all of its suffixes.
 	<Resource source="CF" title="Z Algorithm" url="3107" />
 </Resources>
 
+The Z-algorithm is very similar to KMP, but the z-algorithm uses a different function than $\pi$ and it has an interesting different application than string matching. As I've mentioned
+z-algorithm doesn't use $\pi$ function, instead it uses the **z-function** defined as: $Z_i$ = the length of the longest string that is both the prefix of string $S$ and $S_{i...n}$ - the suffix
+starting from $i$.
+
+Here are the values of the z-function for some strings:
+- aabxaayaab $\rightarrow$ $Z=[0,1,0,0,2,1,0,3,1,0]$
+- aabxaabxcaabx $\rightarrow$ $Z=[0,1,0,0,4,1,0,0,0,8,1,0,0,5,1,0,0]$
```

Comment:
very cool examples here
perhaps explain _why_ it's like so
perhaps "for the first string, z[3] = whatever because ..."

Reply:
ok